services/media/service/src/messaging/register-messaging.ts

```diff
@@ -1,39 +1,348 @@
+    new ImageFailedHandler(storeOutboxMessage, config),
+  ];
+  const commonMessageHandlers: InboxMessageHandler[] = [
+    new DeleteEntityHandler(storeOutboxMessage, config),
+    new CuePointTypesDeclaredHandler(config),
+    new CuePointTypesDeclareFailedHandler(config),
+    new ImageTypesDeclaredHandler(config),
+    new ImageTypesDeclareFailedHandler(config),
+  ];
+  const concurrencyStrategy = createMultiConcurrencyController((message) => {
+    switch (message.messageType) {
+      case MediaServiceMessagingSettings.StartIngest.messageType:
+      case MediaServiceMessagingSettings.StartIngestItem.messageType:
+      case MediaServiceMessagingSettings.UpdateMetadata.messageType:
+      case MediaServiceMessagingSettings.CheckFinishIngestItem.messageType:
+      case MediaServiceMessagingSettings.CheckFinishIngestDocument.messageType:
+      case VideoServiceMultiTenantMessagingSettings
+        .EnsureVideoExistsAlreadyExisted.messageType:
+      case VideoServiceMultiTenantMessagingSettings
+        .EnsureVideoExistsCreationStarted.messageType:
+      case VideoServiceMultiTenantMessagingSettings.EnsureVideoExistsFailed
+        .messageType:
+      case ImageServiceMultiTenantMessagingSettings
+        .EnsureImageExistsAlreadyExisted.messageType:
+      case ImageServiceMultiTenantMessagingSettings
+        .EnsureImageExistsImageCreated.messageType:
+      case ImageServiceMultiTenantMessagingSettings.EnsureImageExistsFailed
+        .messageType:
+        return 'full-concurrency';
+      default: // especially the "common" and "publish" ones are safer with a mutex
+        return 'mutex';
+    }
+  });
+  const [shutdownInSrv] = initializeInboxListener(
+    inboxConfig,
+    [
+      ...publishMessageHandlers,
+      ...ingestMessageHandlers,
+      ...commonMessageHandlers,
+    ],
+    logMapper,
+    {
+      concurrencyStrategy,
+      // Allow the ingest start to be processed a bit longer
+      messageProcessingTimeoutStrategy: (message) =>
+        message.messageType ===
+        MediaServiceMessagingSettings.StartIngest.messageType
+          ? 30_000
+          : 10_000,
+      messageProcessingTransactionLevelStrategy: () =>
+        IsolationLevel.RepeatableRead,
+    },
+  );
+  shutdownActions.push(shutdownInSrv);
+};
+
+const registerRabbitMqMessaging = async (
+  app: Express,
+  ownerPool: OwnerPgPool,
+  config: Config,
+  inboxConfig: InboxConfig,
+  logger: Logger,
+  logMapper: TransactionalLogMapper,
+  shutdownActions: ShutdownActionsMiddleware,
+): Promise<Broker> => {
+  const [storeInboxMessage, shutdownInbox] = initializeInboxMessageStorage(
+    inboxConfig,
+    logMapper,
+  );
+  shutdownActions.push(shutdownInbox);
+  const inboxWriter = new RabbitMqInboxWriter(storeInboxMessage, {
+    // temporary backward compatibility until all your services are updated and all current messages are processed
+    acceptedMessageSettings: [
+      MediaServiceMessagingSettings.StartIngest,
```

Comment:
Does this affect only ingest-related messages, or message types from
`publishingBuilders` and `commonBuilders` should be included as well?